android/src/main/java/com/airbnb/android/react/maps/AirMapView.java

```diff
@@ -273,11 +276,15 @@ public void onHostPause() {
             //noinspection MissingPermission
             map.setMyLocationEnabled(false);
           }
-          paused = true;
+            synchronized (AirMapView.this) {
+                AirMapView.this.onPause();
+                paused = true;
```

Comment:
Can you move setting paused to true and false into onPause and onResume? The boolean logic being here breaks now that you call onPause separately.